ignite/metrics/loss.py

```diff
@@ -33,28 +33,28 @@ class Loss(Metric):
     Attributes:
         required_output_keys: dictionary defines required keys to be found in ``engine.state.output`` if the
             latter is a dictionary. Default, ``("y_pred", "y", "criterion_kwargs")``. This is useful when the
-            criterion function requires additional argument,
-            which can be passed using ``criterion_kwargs``.
+            criterion function requires additional argument, which can be passed using ``criterion_kwargs``.
             See notes below for an example.
 
     Note:
 
         Let's implement a Loss metric that requires ``y_pred``, ``y`` and ``criterion_kwargs`` as input
         for ``criterion`` function. In the example below we show how to setup standard metric like Accuracy
-        and the Loss metric using an ``evaluator`` created
-        with :meth:`~ignite.engine.create_supervised_evaluator` method.
+        and the Loss metric using an ``evaluator`` created with
+        :meth:`~ignite.engine.create_supervised_evaluator` method.
 
         .. code-block:: python
 
             import torch
             import torch.nn as nn
+            from torch.nn.functional import nll_loss
 
             from ignite.metrics import Accuracy, Loss
             from ignite.engine import create_supervised_evaluator
 
             model = ...
 
-            criterion = nn.NLLLoss()
+            criterion = nll_loss()
```

Comment:
I think this is incorrect, as `nll_loss()` is a function executed without any args and will give
```
TypeError: nll_loss() missing 2 required positional arguments: 'input' and 'target'
```

Reply:
Yes, but I tried the above demo code [Passing criterion_kwargs] with nn.NLLLoss() and it gave the error that forward has no parameter "reduction". Any reason for this?  I will change the criterion to `nll_loss` from `nll_loss()`.

Reply:
yes, that's correct, we can do instead `criterion = nll_loss`.

Well, original thought was to provide additional kwargs to some custom loss functions that can require something else. For example a loss that computes on 
```
criterion(y_preds, y, image=x)
```
then we can pass `x` as `criterion_kwargs`. 
The idea is to show that there is no limitation for any kind of custom loss functions.

Reply:
Yes, looking at the whole codebase [at least the Loss, Metric, and Engine parts] I believe that we can achieve this idea very well without much hassle!!